packages/react-components/src/components/Drawer/DrawerMenuItem.tsx

```diff
@@ -66,58 +78,100 @@ export const DrawerMenuItem = (props: _DrawerMenuItemProps): JSX.Element => {
   const onClick = (ev?: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>): void =>
     props.onItemClick && props.onItemClick(ev, props.itemKey);
   const onKeyPress = (ev: React.KeyboardEvent<HTMLElement>): void => onClick && submitWithKeyboard(ev, onClick);
+  const resources = props.reactionResources;
+  const emojiResource: Map<string, string | undefined> = new Map([
+    ['like', resources?.likeReaction?.url],
+    ['heart', resources?.heartReaction?.url],
+    ['laugh', resources?.laughReaction?.url],
+    ['applause', resources?.applauseReaction?.url],
+    ['surprised', resources?.surprisedReaction?.url]
+  ]);
+  const emojis: Reaction[] = ['like', 'heart', 'laugh', 'applause', 'surprised'];
 
   const secondaryIcon = props.secondaryIconProps ? (
     <MenuItemIcon {...props.secondaryIconProps} />
   ) : props.subMenuProps ? (
     <MenuItemIcon iconName="ChevronRight" />
   ) : undefined;
 
-  return (
-    <FocusZone shouldFocusOnMount={props.shouldFocusOnMount}>
+  if (props.itemKey === 'reactions') {
```

Comment:
Instead of bolting this into the existing DrawerMenuItem and coupling a drawer menu with a specific use case, could this be its own component `ReactionsDrawerMenuItem`

Reply:
Thanks for the suggestion and guidance. :)